### ISSUE TITLE
Display of BIT types broken in 1.1

### ISSUE THREAD

**Issue by sanderv** [sequelpro/sequelpro#2254]
Boolean handling in SP1.1 seems buggy. See screenshot. Sorting on a BIT field gives mixed booleans, not sorted. A value that is set to true shows up as '000000000', false as '10000000'.

![screen shot 2015-10-01 at 15 36 50](https://cloud.githubusercontent.com/assets/734537/10222072/502dfd88-6852-11e5-80af-98ebeb11ac6c.png)

Querying the database:

    SELECT IF (sentDigitally, "y", "n'")
    FROM printbill
    WHERE id = 23197949;

does work and yields correct y/n values.

This query

    SELECT id, sentDigitally, IF (sentDigitally, "j", "n")
    FROM printbill
    ORDER BY id DESC
    LIMIT 1000

Gives this result:
![screen shot 2015-10-01 at 15 44 41](https://cloud.githubusercontent.com/assets/734537/10222271/61aff86c-6853-11e5-94a0-0ba9f93f0d20.png)

So, alternating display of the 'native' boolean where the IF-function shows all fields are false.

**Comment by dmoagx** [comment]
Can reproduce the issue in 1.1
It looks like the right-most bit somehow gets moved to be the left-most bit.

**Comment by dmoagx** [comment]
```sql
mysql> SHOW CREATE TABLE bits;

CREATE TABLE `bits` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `bitf` bit(3) NOT NULL DEFAULT b'0',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=6 DEFAULT CHARSET=latin1
```
```
mysql> SELECT id,HEX(bitf) FROM bits;
+----+-----------+
| id | HEX(bitf) |
+----+-----------+
|  1 | 1         | // b'1'
|  2 | 0         | // b'0'
|  3 | 5         | // b'101'
|  4 | 2         | // b'10'
|  5 | 1         | // b'01'
+----+-----------+
```
![bildschirmfoto 2015-10-01 um 17 48 54 589](https://cloud.githubusercontent.com/assets/5793864/10225813/b8a2db6a-6864-11e5-9b73-c6d15d7f942f.png)

**Comment by dmoagx** [comment]
This was caused by 19bde59d3eae295d907e41a4185cd487ae780123
Stuart probably overlooked in which order the commands were executed.
Namely:
```
calculate value with i=0 (right side expression)
increment i (++i)
assign to field (left side expression)
```
While it is now:
```
increment i (++i)
calculate value with i=1 (right side expression)
assign to field (left side expression)
```

**Comment by dmoagx** [comment]
Since I can image how this will continue, let's try something different.

**Comment by geniuscd** [comment]
I have two columns of type BIT(1) where all their rows are '0' in an InnoDB Table.
![3](https://cloud.githubusercontent.com/assets/3702631/10275492/986eb4a6-6b50-11e5-8796-babb25076d1d.png)
but the UI is showing different value, sometimes it shows '100000'  as  if the value is 1, but its not.
![1](https://cloud.githubusercontent.com/assets/3702631/10275500/a3e39810-6b50-11e5-87c7-79de681858df.png)
even when i'm running a select statement it shows differently.
![2](https://cloud.githubusercontent.com/assets/3702631/10275502/a77b9018-6b50-11e5-85a3-1051b9efe4a3.png)

running v1.1 on OSX 10.9.2 (mysql 5.1 on cents 6.4 server)

**Comment by krlbaker** [comment]
Same problem with my database and v1.1

**Comment by kasperlabela** [comment]
Same here, very inconvenient

**Comment by dmoagx** [comment]
Until 1.1.1 is released either downgrade to 1.0.2 (http://www.sequelpro.com/release-archive) or try an experimental nightly build which has this fixed: http://nightly.sequelpro.com/

**Comment by sanderv** [comment]
Thanks, good to see that bugs are handled this quick!

**Comment by Enzo90910** [comment]
I cannot provide proof but I have a strong suspicion that a variant of this bug was also present in exported .sql files:
- all bits at 1 are exported as 10000000 instead of 00000001
- sometimes bit which are actually at 1 are exported as 0.
Just wanted to point it out so that someone can check it was fixed too.

**Comment by dmoagx** [comment]
Yes, see #2262

**Comment by jstranders** [comment]
I have the same. Please fix.

**Comment by rowanbeentje** [comment]
@jstranders As you can see above, the bugs are fixed and awaiting a new release; in the meantime you can either downgrade to 1.0.2 or download a nightly, see the links above.

**Comment by nikyoudale** [comment]
How long until 1.1.1 is released? It would be awesome to have a fix for this issue.

**Comment by sgarbesi** [comment]
+1 major bug

**Comment by prior** [comment]
+1 major bug

**Comment by esimioni** [comment]
+1 For GA release with this fix ASAP.

**Comment by leighwhitney** [comment]
I'm happy to find this thread - I thought I was going crazy (where a boolean sometimes shows as true but is in fact false). 

+1 from me as well.

**Comment by seybsen** [comment]
+1, I miss sequel pro

**Comment by menicosia** [comment]
Annnnd a :+1: from me as well, just hit this bug as I upgraded to 1.1 this morning. :(

**Comment by rbagchi** [comment]
+1 This is really problematic since ssh tunnels don't work on the old
version after upgrading to El Capitan

**Comment by karabfak** [comment]
+1 from me as well, have been dealing with this the last 3 months and its quite annoying

**Comment by chbuck** [comment]
+1 from me
I discovered, the shown bit value depends on the value in the previous column.

**Comment by davidnortonjr** [comment]
:+1:  please release 1.1.1

**Comment by adhesivewombat** [comment]
+1 from me too, bit column seems to be dependent on the column to the left of it.

**Comment by watson-song** [comment]
where is 1.1.1, this issue really makes me crazy, switch to workbench then find the problem, the boolean value is not set correctly, but sequelpro always show me wrong number.......

**Comment by GitHubMurray** [comment]
+1 from me with v1.1

**Comment by eloan** [comment]
+1 from me too (v1.1)

**Comment by ctandy** [comment]
+1 from me too. Makes testing very difficult.

**Comment by EmmEm** [comment]
+1 v1.1.0.1

**Comment by boblepepeur** [comment]
+1, i love this tool !

**Comment by KyotoFox** [comment]
+1

**Comment by romain67** [comment]
+1 v1.1.0.1

**Comment by mathifonseca** [comment]
+1 for a whole company!

**Comment by hansregeer** [comment]
+1 have to fallback to workbench now...

**Comment by graymouser** [comment]
+1  - v1.1.0.1

**Comment by underkin** [comment]
+1

**Comment by dmoagx** [comment]
# This is an issue tracker, not a Google+ page. 

If you do not have to add anything productive, do not comment.

**Comment by dmoagx** [comment]
Btw 1.1.1 RC1 is out now and should fix this issue:

https://github.com/sequelpro/sequelpro/releases/tag/1.1.1-rc1
(use the RC1a download)

**Comment by cadecannon** [comment]
Just ran into this same issue with a BIT field in a legacy db and 1.1.1 RC1a has solved the issue for me.

**Comment by dmoagx** [comment]
Issue fixed in 1.1.1
See http://www.sequelpro.com/blog/2016.02/sequel-pro-1-1-1/
